swaybar/render.c

```diff
@@ -355,17 +355,8 @@ static const char *strip_workspace_number(const char *ws_name) {
 	return ws_name;
 }
 
-static const char *strip_workspace_name(const char *ws_name) {
-	size_t len = strlen(ws_name);
-	for (size_t i = 0; i < len; ++i) {
-		if (ws_name[i] < '0' || ws_name[i] > '9') {
-			if (':' == ws_name[i] && i < len - 1 && i > 0) {
-				return &ws_name[0];
-			}
-			return ws_name;
-		}
-	}
-	return ws_name;
+static const char *strip_workspace_name(char *ws_name) {
+	return strtok(ws_name, ":");
```

Comment:
This is likely a poor way of doing it, but I am not familiar enough with C to know a better way...

Reply:
`strtok` allocates a new string, unlike `strip_workspace_number` which returns a pointer within the same string. As it stands there's a memory leak here.

I note this is being allocated while rendering, so I think a different approach needs to be taken. We could add a new property to the `swaybar_workspace` struct for the renderable name (ie. stripped), then update it when we receive workspace events. Then render using that property.

Reply:
Should I add a `ws_number` and `ws_name` field to `swaybar_workspace` then? I assume they'd both need to be `char *`.